initial github_traffic script commit.

milljm opened the pull request:
Initial commit. This will require an update to our moose-environment package to include the python-mechanize module for seamless use. Refs #5049

Commit:
initial github_traffic script commit. Refs #5049

moosebuild commented:
Results of testing 383391e3ed232196f6c25a3bd19831cb2696d8e5 using moose_PR_pre_check recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/15788

dschwen commented:
Do we really need our users to run this?

jwpeterson commented:
No, but we wanted to make the script publicly-available.

permcody commented on scripts/github_traffic.py:
```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+import cookielib
```
Mmm - cookie

permcody commented on scripts/github_traffic.py:
```diff
@@ -0,0 +1,86 @@
+  br = mechanize.Browser()
+
+  # Cookie Jar
+  cj = cookielib.LWPCookieJar()
```
Mmm - CookieJar!

permcody commented on scripts/github_traffic.py:
```diff
@@ -0,0 +1,86 @@
+
+def verifyArgs(args):
+  if args.repo is None or len(args.repo.split('/')) != 2:
+    print '\nYou must specify a repository you are insterested in scrapeing:\n\t --repo foo/bar\n\nNote: GitHub is case-sensitive, so your arguments must be too'
```
Type - "scraping"

moosebuild commented:
Results of testing 383391e3ed232196f6c25a3bd19831cb2696d8e5 using moose_PR_app_tests recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/15791

moosebuild commented:
Results of testing 2cce8a0358337b96b8d6690f91908b47bed41a4a using moose_PR_pre_check recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/15796

moosebuild commented:
Results of testing 383391e3ed232196f6c25a3bd19831cb2696d8e5 using moose_PR_test recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/15789

moosebuild commented:
Results of testing 2cce8a0358337b96b8d6690f91908b47bed41a4a using moose_PR_app_tests recipe:

Failed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/15799

moosebuild commented:
Results of testing 2cce8a0358337b96b8d6690f91908b47bed41a4a using moose_PR_test recipe:

Failed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/15797

moosebuild commented:
Results of testing 2cce8a0358337b96b8d6690f91908b47bed41a4a using moose_PR_test_dbg recipe:

Failed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/15798

dschwen commented:
How could this possibly fail?

Commit:
Add short arguments.
Ask for password immediately rather than waiting for login page to load.
Handle case where USER is not available in the environment.
Refs #5049

moosebuild commented:
Results of testing fe890f95b483d8fa1673181175416d83fe6a207d using moose_PR_pre_check recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/15800

jwpeterson commented:
> How could this possibly fail?

Accidental libmesh de-update

moosebuild commented:
Results of testing fe890f95b483d8fa1673181175416d83fe6a207d using moose_PR_test recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/15801

moosebuild commented:
Results of testing fe890f95b483d8fa1673181175416d83fe6a207d using moose_PR_app_tests recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/15803

moosebuild commented:
Results of testing 383391e3ed232196f6c25a3bd19831cb2696d8e5 using moose_PR_test_dbg recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/15790

moosebuild commented:
Results of testing fe890f95b483d8fa1673181175416d83fe6a207d using moose_PR_test_dbg recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/15802
